lib/components/wallet_connect/approve_token_transfer.dart

```diff
@@ -0,0 +1,269 @@
+        toIdName = item.first.name;
+      });
+    }
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
+
+  List<Widget> getButtons() {
+    final l10n = l10nOf(context);
+
+    return [
+      SizedBox(
+        width: double.infinity,
+        height: ButtonStyles.buttonHeight,
+        child: FilledButton(
+            style: FilledButton.styleFrom(
+              backgroundColor: LightThemeColors.primary40,
+            ),
+            onPressed: isLoading
+                ? null
+                : () async {
+                    //Authenticate the user
+                    if (mounted) {
+                      bool authenticated = await reAuthDialog(context);
+                      if (!authenticated) {
+                        if (mounted) {
+                          Navigator.pop(context);
+                        }
+                        return;
+                      }
+                    }
+                    setState(() {
+                      isLoading = true;
+                    });
+                    //Get current tick
+                    try {
+                      int latestTick = (await _liveApi.getCurrentTick()).tick;
+                      int targetTick = latestTick + defaultTargetTickType.value;
+
+                      //Send the transaction to backend
+                      bool result = false;
+                      if (mounted) {
+                        //TODO ERROR HANDLING
+                        result = await sendTransactionDialog(
+                            context,
+                            widget.fromID!,
+                            widget.toID!,
+                            widget.amount,
+                            targetTick);
+
+                        if (!result) {
+                          throw "Could not send transaction";
+                        }
+
+                        setState(() {
+                          isLoading = true;
+                        });
+                        //If the transaction was successful
+
+                        if (mounted) {
+                          Navigator.of(context).pop(ApproveTokenTransferResult(
+                              //Return the success and tick
+
+                              tick: targetTick));
+                          getIt.get<PersistentTabController>().jumpToTab(1);
+                          getIt<GlobalSnackBar>().show(
+                              l10nOf(context) //Show snackbar
+                                  .generalSnackBarMessageTransactionSubmitted(
+                                      targetTick.toString()));
+                        }
+                      }
+                    } catch (e) {
+                      setState(() {
+                        isLoading = false;
+                      });
+
+                      if (mounted) {
+                        Navigator.of(context).pop(ApproveTokenTransferResult(
+                            //Return the success and tick
```

Comment:
@Qubic-Hub  why this would be success? I see there is a throw in the try block when !result

Reply:
@ahmed-tarek-salem  could you please have a look on this?

Reply:
Fixed with [304eaf0](https://github.com/qubic/wallet-app/pull/110/commits/304eaf0bba185872db126a8a5aabe3b43a1533df)